extensions/iam/decentralized-identity/identity-did-crypto/src/main/java/org/eclipse/dataspaceconnector/iam/did/crypto/credentials/VerifiableCredentialFactory.java

```diff
@@ -121,61 +86,48 @@ public static SignedJWT create(PrivateKeyWrapper privateKey, Map<String, String>
      * @param publicKey The claiming party's public key, passed as a {@link PublicKeyWrapper}
+     * @param audience  The intended audience
      * @return true if verified, false otherwise
      */
-    public static boolean verify(SignedJWT verifiableCredential, PublicKeyWrapper publicKey) {
+    public static Result<Void> verify(SignedJWT jwt, PublicKeyWrapper publicKey, String audience) {
+        // verify JWT signature
         try {
-            return verifiableCredential.verify(publicKey.verifier());
+            var verified = jwt.verify(publicKey.verifier());
+            if (!verified) {
+                return Result.failure("Invalid signature");
+            }
         } catch (JOSEException e) {
-            throw new CryptoException(e);
+            // Unable to verify, e.g. the JWS algorithm is not supported
+            return Result.failure(e.getMessage());
         }
-    }
 
-    /**
-     * Verifies a VerifiableCredential using the issuer's public key
-     *
-     * @param verifiableCredential a {@link SignedJWT} that was sent by the claiming party.
-     * @param publicKeyPemContent The claiming party's public key, i.e. the contents of the public key PEM file.
-     * @return true if verified, false otherwise
-     */
-    public static boolean verify(SignedJWT verifiableCredential, String publicKeyPemContent) {
+        JWTClaimsSet jwtClaimsSet;
         try {
-            var key = ECKey.parseFromPEMEncodedObjects(publicKeyPemContent);
-            return verify(verifiableCredential, (ECKey) key);
-        } catch (JOSEException e) {
-            throw new CryptoException(e);
+            jwtClaimsSet = jwt.getJWTClaimsSet();
+        } catch (ParseException e) {
+            // The payload of the JWT doesn't represent a valid JSON object and a JWT claims set
+            return Result.failure(e.getMessage());
         }
 
-    }
+        // verify claims
+        var exactMatchClaims = new JWTClaimsSet.Builder()
+                .audience(audience)
+                .subject(VERIFIABLE_CREDENTIAL)
+                .build();
+        var requiredClaims = Set.of(
+                ISSUER_CLAIM,
+                EXPIRATION_TIME_CLAIM);
 
-    /**
-     * Parses a {@link SignedJWT} back to a Java object from its serialized form.
-     *
-     * @param jwtString The serialized form of the {@code SignedJWT}, which can be generated using {@link SignedJWT#serialize()}.
-     * @return a {@link SignedJWT} containing the decoded information
-     */
-    public static SignedJWT parse(String jwtString) {
+        var claimsVerifier = new DefaultJWTClaimsVerifier<>(exactMatchClaims, requiredClaims);
         try {
-            return SignedJWT.parse(jwtString);
-        } catch (ParseException e) {
-            throw new CryptoException(e);
+            claimsVerifier.verify(jwtClaimsSet);
+        } catch (BadJWTException e) {
+            // claim verification failed
```

Comment:
same as above

Reply:
This is already in the message produced by nimbus, e.g. `JWT missing required claims: [aud]`, `JWT "aud" claim has value [https://otherserver.com] but should be [test-audience]`